enhancements/security/openshift-image-policy.md

```diff
@@ -0,0 +1,700 @@
+
+I think this risk is acceptable, because the installer knob is supposed to be for internal testing of short-lived clusters.
+But if it turns out we need something in-cluster to migrate these settings, we may want to pivot to [have the machine-config operator manage the policy directly](#have-the-machine-config-operator-manage-the-policy-directly) with a more structured in-cluster knob.
+
+#### Transitioning unsigned-release testing
+
+There is a transitional cost to the additional step (`OPENSHIFT_INSTALL_EXPERIMENTAL_DISABLE_IMAGE_POLICY` for installs, `overrides` for updating older clusters) needed to run an unsigned release image for internal testing.
+But that transitional cost is only expected to affect internal use-cases; external users should always be using signed releases.
+
+#### Restricted-network use
+
+Disconnected/restricted-network use is out of scope for this enhancement, so external clusters running `TechPreviewNoUpgrade` and getting this feature may also need the `OPENSHIFT_INSTALL_EXPERIMENTAL_DISABLE_IMAGE_POLICY` or `overrides` workaround.
+It's possible that an ImageDigestMirrorSet like this would be sufficient:
+
+```yaml
+apiVersion: config.openshift.io/v1
+kind: ImageDigestMirrorSet
+metadata:
+  name: openshift-release-payload
+spec:
+  imageDigestMirrors:
+  - mirrors:
+    - example.com/my-mirror/whatever
+    source: quay.io/openshift-release-dev/ocp-release
+```
+
+In any case, it's out of scope for a reason, but until it gets picked up, it might make things more difficult for `TechPreviewNoUpgrade` disconnected/restricted-network clusters.
+
+##### Mirror registries that do not support Sigstore signatures
+
+As a subset of the restricted-network use case, when a user has an ImageDigestMirrorSet set to point at a mirror, and the canonical quay.io is either not reachable, or [`mirrorSourcePolicy: NeverContactSource`][ImageDigestMirrorSet] is set, the mirror registry will need to allow Sigstore signatures in order for CRI-O to be able to retrieve signatures.
+Quay support for Sigstore is [new and ongoing][PROJQUAY-3400], so it's likely that there are other mirror registries in use where Sigstore hosting is not yet possible.
+But again, restricted-network use is out of scope for this initial policy, and once it comes into scope, mirror support will be a generic issue for all ClusterImageProfile and ImageProfile users, and will not be unique to OpenShift's internal policies.
+
+#### Security
+
+How will security be reviewed and by whom?
+
+I have no idea; suggestions welcome :).
+
+There will be a gap during install before the ClusterImagePolicy has been created and rolled out to nodes, during which unsigned `quay.io/openshift-release-dev/ocp-release` images might run.
+But we do not perform any in-cluster signature checking during installs today; they are secured by a trusted installer using a baked-in release image digest.
```

Comment:
We would want this documented.

Reply:
It's hard to fix, because we get the ClusterImagePolicy from the release image, so you need that first release image pull to go through before you know what the policy should be.  You could pull the policy out of the release image and inject it into the installer binary at extraction time, like we [currently][1] [do][2] for other release-image metadata.  But Sigstore signatures on release images are still too late to establish trust in the installer.   And once you have established trust in the installer, the installer's injected-at-extraction-time by-digest release image pullspec establishes trust in the release image.  So it's not clear to me what gap earlier Sigstore signature coverage would close.

[1]: https://github.com/openshift/installer/blame/a651e1a49ceafcb5cbe6f6244e315a2381013aab/pkg/version/version.go#L44-L56
[2]: https://github.com/openshift/oc/blob/110cba7f25fd1c53017f5c207614b8d88fb97dd1/pkg/cli/admin/release/extract_tools.go#L883-L905

Reply:
I’m sorry, that’s my mistake; it says “there will be a gap during install”, I was reading that as “after install there is a gap where an upgrade could be triggered before a policy has finished rolling out”.

I agree that the digest references flowing from the installer through the release image to the various product images is sufficient; and I don’t think that particularly needs documentation (end-user-focused documentation, at least).